fastlane/Fastfile

```diff
@@ -281,3 +278,24 @@ before_all do |lane|
   end
   # rubocop:enable Style/IfUnlessModifier
 end
+
+def compute_release_branch_name(options:, version: release_version_current)
+  branch_option = :branch
+  branch_name = options[branch_option]
+
+  if branch_name.nil?
+    branch_name = release_branch_name(version:)
+    UI.message("No branch given via option '#{branch_option}'. Defaulting to #{branch_name}.")
+  end
+
+  branch_name
+end
+
+def release_branch_name(version: release_version_current)
+  "release/#{version}"
+end
+
+def ensure_git_branch_is_release_branch
+  # Verify that the current branch is a release branch. Notice that `ensure_git_branch` expects a RegEx parameter
+  ensure_git_branch(branch: '^release/')
```

Comment:
Noted. I didn't run into this issue during my tests, #22194, but maybe it was due to some additional test setup...

However, this might explain something that surprised me and that I filed for investigation. In the screenshot below from [this build](https://buildkite.com/automattic/wordpress-ios/builds/18903#018c4208-9140-4696-9590-a5c1e12c4715) the `git_branch` output is the branch the lane started on, not the new one.

![image](https://github.com/wordpress-mobile/WordPress-iOS/assets/1218433/ddc6e42d-a6c8-4a57-9f3f-6c1f6bd0b21b)

Addressed in d9691c6ec19f3abfe133e58e8f33b6f9a0d9d917

Reply:
Yeap, that's the exact issue that environment variable addresses. If you'd like to learn more about it, [here is the PR](https://github.com/fastlane/fastlane/pull/21597) that we submitted to `fastlane`.